packages/manager/apps/pci-ai-dashboard/src/pages/dashboard/datastore/_components/AddDatastore.modal.tsx

```diff
@@ -169,11 +147,11 @@ const AddDatastore = ({
                   <FormLabel>{t('formAddDatastoreFieldRegionLabel')}</FormLabel>
                   <FormControl>
                     <Select value={field.value} onValueChange={field.onChange}>
-                      <SelectTrigger>
+                      <SelectTrigger data-testid="select-region-trigger">
                         <SelectValue />
                       </SelectTrigger>
                       <SelectContent>
-                        {regions.map((region) => (
+                        {regionsQuery.data.map((region) => (
```

Comment:
The ìsLoading`attribute on RouteModal does not avoid first rendering, we have to check if data is not undefined.
```suggestion
                        {regionsQuery.data?.map((region) => (
```